module/zfs/vdev_rebuild.c

```diff
@@ -307,6 +309,14 @@ vdev_rebuild_complete_sync(void *arg, dmu_tx_t *tx)
 	vdev_rebuild_phys_t *vrp = &vr->vr_rebuild_phys;
 
 	mutex_enter(&vd->vdev_rebuild_lock);
+
+	/* Handle the lost reset wanted signal */
+	if (vd->vdev_rebuild_reset_wanted) {
+		mutex_exit(&vd->vdev_rebuild_lock);
+		vdev_rebuild_reset_sync((void *)(uintptr_t)vd->vdev_id, tx);
```

Comment:
We could make this even a little simpler since the passed `arg` is the top level vdev id.

```suggestion
		vdev_rebuild_reset_sync(arg, tx);
```